From 1d23011d66f7aece19cc0c375a1de8440acc3b7d Mon Sep 17 00:00:00 2001
From: AKASHI Takahiro <takahiro.akashi@linaro.org>
Date: Tue, 21 Apr 2020 09:37:52 +0900
Subject: [PATCH] lib/crypto, efi_loader: avoid multiple inclusions of header
 files

By adding extra symbols, we can now avoid including x509_parser and
pkcs7_parser.h files multiple times.

Suggested-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
Signed-off-by: AKASHI Takahiro <takahiro.akashi@linaro.org>
Don't include include x509_parser.h twice.
Reviewed-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/crypto/pkcs7_parser.h | 4 ++++
 lib/crypto/x509_parser.h  | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/lib/crypto/pkcs7_parser.h b/lib/crypto/pkcs7_parser.h
index 6565fdc2d4..b8234da45a 100644
--- a/lib/crypto/pkcs7_parser.h
+++ b/lib/crypto/pkcs7_parser.h
@@ -5,6 +5,9 @@
  * Written by David Howells (dhowells@redhat.com)
  */
 
+#ifndef _PKCS7_PARSER_H
+#define _PKCS7_PARSER_H
+
 #include <linux/oid_registry.h>
 #include <crypto/pkcs7.h>
 #include "x509_parser.h"
@@ -63,3 +66,4 @@ struct pkcs7_message {
 	size_t		data_hdrlen;	/* Length of Data ASN.1 header */
 	const void	*data;		/* Content Data (or 0) */
 };
+#endif /* _PKCS7_PARSER_H */
diff --git a/lib/crypto/x509_parser.h b/lib/crypto/x509_parser.h
index c233f136fb..4cbdc1d661 100644
--- a/lib/crypto/x509_parser.h
+++ b/lib/crypto/x509_parser.h
@@ -5,6 +5,9 @@
  * Written by David Howells (dhowells@redhat.com)
  */
 
+#ifndef _X509_PARSER_H
+#define _X509_PARSER_H
+
 #include <linux/time.h>
 #include <crypto/public_key.h>
 #include <keys/asymmetric-type.h>
@@ -55,3 +58,4 @@ extern int x509_decode_time(time64_t *_t,  size_t hdrlen,
  */
 extern int x509_get_sig_params(struct x509_certificate *cert);
 extern int x509_check_for_self_signed(struct x509_certificate *cert);
+#endif /* _X509_PARSER_H */
-- 
2.26.2

